### DISCUSSION TITLE
npm config module fails in production vercel build with no configurations found

### DISCUSSION THREAD

**Discussion by ghost** [vercel/community#1093] [Help] [answered]
I am using the npm config package locally with a config/default.json file at the project root. It works great when used locally. When building in Vercel, I get this error:

> 2022-11-25T18:30:53.027Z	80ce2123-760a-4b72-8cbc-26d60beb9e65	ERROR	WARNING: NODE_ENV value of 'production' did not match any deployment config file names.
> 2022-11-25T18:30:53.027Z	80ce2123-760a-4b72-8cbc-26d60beb9e65	ERROR	WARNING: See https://github.com/node-config/node-config/wiki/Strict-Mode
> 2022-11-25T18:30:53.027Z	80ce2123-760a-4b72-8cbc-26d60beb9e65	ERROR	WARNING: No configurations found in configuration directory:/var/task/config
> 2022-11-25T18:30:53.027Z	80ce2123-760a-4b72-8cbc-26d60beb9e65	ERROR	WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.

Then all my config values are undefined. It looks like for some reason the /var/task/config does not have my default.json file in it.

**Comment by amyegan** [comment, marked as answer]
Hi @sharpsteelsoftware1! Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a closer look at the problem?

**Reply by ghost** [reply]
https://github.com/sharpsteelsoftware1/nextjs-config-minimal-repro

**Reply by amyegan** [reply]
The error seems to be a problem with `node-config`. I was able to reproduce it with a local build. I also tried downgrading to Next.js 12 and ran into the same problem. You can learn more about the error at https://github.com/node-config/node-config/wiki/Strict-Mode.

😞 I don't know enough about `node-config` to be able to offer guidance on config changes to get it working. You should ask the [`node-config` experts](https://github.com/node-config/node-config/discussions) for advice. They know it best and probably already know how to solve this problem.

**Reply by ghost** [reply]
Thank you Amy! I dropped config and moved onward. But I wanted to alert Vercel so that you had a chance to look into this and consider resolving it for your users as config is a major npm package. Thank you for your help.
